src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-details/rgw-multisite-details.component.ts

```diff
@@ -378,4 +492,45 @@ export class RgwMultisiteDetailsComponent implements OnDestroy, OnInit {
       });
     }
   }
+
+  enableRgwModule() {
+    let $obs;
+    const fnWaitUntilReconnected = () => {
+      observableTimer(2000).subscribe(() => {
+        // Trigger an API request to check if the connection is
+        // re-established.
+        this.mgrModuleService.list().subscribe(
+          () => {
+            // Resume showing the notification toasties.
+            this.notificationService.suspendToasties(false);
+            // Unblock the whole UI.
+            this.blockUI.stop();
+            // Reload the data table content.
+          },
+          () => {
+            fnWaitUntilReconnected();
+          }
+        );
+      });
+    };
+
+    if (!this.rgwModuleStatus) {
+      $obs = this.mgrModuleService.enable('rgw');
+    }
+    $obs.subscribe(
+      () => undefined,
+      () => {
+        // Suspend showing the notification toasties.
+        this.notificationService.suspendToasties(true);
+        // Block the whole UI to prevent user interactions until
+        // the connection to the backend is reestablished
+        this.blockUI.start($localize`Reconnecting, please wait ...`);
+        fnWaitUntilReconnected();
+        setTimeout(() => {
+          this.notificationService.show(NotificationType.success, $localize`Enabled RGW Module`);
+          this.ngOnInit();
+        }, 10000);
+      }
+    );
+  }
```

Comment:
Personally I prefer to use the radosgw-admin cli to do the import/export. Since you are going with this approach, we can go for it now. But this will be broken in ceph-dev (and ideally it shouldn't be).